frontend/leaderboard/src/Sheets.re

```diff
@@ -121,4 +121,83 @@ let createClient = (clientCredentials, cb) => {
       }
     });
   };
+};
+
+let updatePointsColumns = (usernamePointsMap, sheetsData, columnIndex) => {
+  // Iterate through all the rows in the fetched Google Sheets data
+  Array.iter(
+    row => {
+      row
+      |> Array.iter(rowValue =>
```

Comment:
The reasoning for the inner Array.iter is because I'm making the assumption that the username field can be any column. Because of this, I loop through the row until a value is matched with the username map. 

Using that information, grabbing the value from the map and value from the sheets cell is not tied to the username. I ran the script with usernames of only digits, random characters, and only alphabetic characters and the output seems to be consistent. 

Another approach that I thought while developing this would be keeping track of the username column when it's found while calculating challenges and passing that into the function to guarantee proper indexing. Maybe we can talk further?